lib/tasks/after_tasks.rake

```diff
@@ -740,6 +740,34 @@ namespace :After do
     WorkIndexer.create_mapping
   end
 
+  desc "Fix tags with extra spaces"
+  task(fix_tags_with_extra_spaces: :environment) do
+    total_tags = Tag.count
+    total_batches = (total_tags + 999) / 1000
+    puts "Inspecting #{total_tags} tags in #{total_batches} batches"
+
+    report_string = "Tag ID,Old tag name,New tag name\n"
+    Tag.find_in_batches.with_index do |batch, index|
+      batch_number = index + 1
+      progress_msg = "Batch #{batch_number} of #{total_batches} complete"
+
+      batch.each do |tag|
+        next unless tag.name != tag.name.squish
+
+        old_tag_name = tag.name
+        new_tag_name = old_tag_name.gsub(/[[:space:]]/, "_")
+
+        tag.update_column(:name, new_tag_name)
+
+        report_row = [tag.id, old_tag_name, new_tag_name].join(",") + "\n"
```

Comment:
Does this work for tags with quotes in them? I know quotation marks are a special character for some CSV implementations.

Reply:
This will escape double quotes as per ruby standard by prefixing with a backslash: `\"`

CSV handling may require double double-quotes (depending on the program), but is that something we want to do in the task here, or just leave it up to find and replace for whoever ends up needing to open the CSV?

Reply:
Since it's program-dependent, I think we can leave it for whoever needs to open the file.

Reply:
I just tested it, and it doesn't escape quotes.

`to_csv` does the right thing and would save the people downstream some trouble.

Reply:
You're right of course.  I was doing string matching with stdout and didn't double escape.